crates/bevy_text/src/text.rs

```diff
@@ -70,12 +102,87 @@ pub struct TextSection {
     pub style: TextStyle,
 }
 
+impl TextSection {
+    /// Create a new [`TextSection`].
+    pub fn new(text: impl Into<String>, style: TextStyle) -> Self {
+        Self {
+            value: text.into(),
+            style,
+        }
+    }
+
+    /// Create an empty [`TextSection`] from a style. Useful when the text will be set dynamically.
+    pub fn from_style(style: TextStyle) -> Self {
+        Self {
+            value: Default::default(),
+            style,
+        }
+    }
+}
+
 #[derive(Debug, Clone, Copy, Reflect)]
 pub struct TextAlignment {
     pub vertical: VerticalAlign,
     pub horizontal: HorizontalAlign,
 }
 
+impl TextAlignment {
```

Comment:
Minor note here: We should keep in mind that fixing things to "left" and "right" can be disadvantageous when localizing to RTL languages, because then you'll likely want to have it in the other direction.

This is why CSS flexbox talks about "start" and "end" instead, which then depend on the writing mode ([Docs](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout/Basic_Concepts_of_Flexbox#start_and_end_lines)).
Of course this also has drawbacks, such as being less intuitive for beginners and introducing additional internal complexity (the writing mode must be specified or automatically recognized).

So I'm not sure what the best approach would be here, but I think it's a good idea to keep RTL in mind so that we have less work in the future :)

Reply:
Having the constants pre-defined like this is definitely nice though!

Reply:
Solving this will likely have to do with establishing what should be dependent or independent from the writing direction. 

Either the user specifies "align to main start" and the writing directions puts it to the top, bottom, left and right depending on writing direction specified. 

Or the user specifies "align top left" and the system figures out how to place it there regardless of the writing directions used. 

My gut feeling tells me that this really should be handled upstream in taffy, and end users should never have to worry about concepts like `main` and `cross`, but not fully sure this is the correct approach.

Reply:
This is a tricky issue.
For straightforward boxes of text (eg. an entry in a players quest journal) changing the text alignment based on the language makes sense, but games tend to have more complex UI where text and images are positioned based on other factors like stylistic choices. Changing the text alignment globally would likely break a lot of things (eg. hud-elements).
It's likely impossible to handle this automatically. Localisation is hard.

I'm going to mark this convo as resolved though, as this is beyond this PR :)